libs/hwes/event.go

```diff
@@ -40,6 +49,28 @@ func NewEventWithData(aggregate Aggregate, eventType string, data interface{}) (
 	return event, nil
 }
 
+// EventWithUserID injects the UserID from the passed context via common.GetUserID().
+// If no UserID was injected, prior to this function call, an error will be returned.
+// Make sure to inject the UserID via a Middleware in the API layer.
+func EventWithUserID(ctx context.Context, event Event) (Event, error) {
```

Comment:
Make this the "normal" Event constructor, and rename the other to "EventWithoutUserID", as most events (if not all?) are initiated by a user.

Reply:
maybe use the options pattern instead of creating multiple constructors?